examples/source/personalization/OAuth2_Login/api.js

```diff
@@ -0,0 +1,298 @@
+
+// Endpoint URLs with corresponding headers and key for each provider
+const USERINFO_ENDPOINT = {
+  google: {
+    url: 'https://www.googleapis.com/oauth2/v3/userinfo',
+    headers: (token) => {
+      return `Bearer ${token}`;
+    },
+    propertyName: 'name',
+  },
+  github: {
+    url: 'https://api.github.com/user',
+    headers: (token) => {
+      return `token ${token}`;
+    },
+    propertyName: 'login',
+  },
+  facebook: {
+    url: 'https://graph.facebook.com/v3.3/me',
+    headers: (token) => {
+      return `Bearer ${token}`;
+    },
+    propertyName: 'name',
+  },
+};
+
+// Provider configurations
+const GOOGLE_CONFIG = {
+  provider: 'google',
+  OAUTH_ID_KEY: 'google_client_id',
+  OAUTH_SECRET_KEY: 'google_client_secret',
+  authCodeUrl: (oauthConfig) => {
+    return `https://accounts.google.com/o/oauth2/v2/auth?response_type=code&client_id=${oauthConfig.id}&scope=openid profile&state=${oauthConfig.state}`;
+  },
+  tokenUrl: (oauthConfig) => {
+    return {
+      url: `https://accounts.google.com/o/oauth2/token?grant_type=authorization_code&client_id=${oauthConfig.id}&client_secret=${oauthConfig.secret}`,
+      options: {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+      },
+    };
+  },
+};
+const GITHUB_CONFIG = {
+  provider: 'github',
+  OAUTH_ID_KEY: 'github_client_id',
+  OAUTH_SECRET_KEY: 'github_client_secret',
+  authCodeUrl: (oauthConfig) => {
+    return `https://github.com/login/oauth/authorize?client_id=${oauthConfig.id}&state=${oauthConfig.state}`;
+  },
+  tokenUrl: (oauthConfig) => {
+    return {
+      url: `https://github.com/login/oauth/access_token?client_id=${oauthConfig.id}&client_secret=${oauthConfig.secret}`,
+      options: {
+        method: 'post',
+        headers: {
+          accept: 'application/json',
+        },
+      },
+    };
+  },
+};
+const FACEBOOK_CONFIG = {
+  provider: 'facebook',
+  OAUTH_ID_KEY: 'facebook_client_id',
+  OAUTH_SECRET_KEY: 'facebook_client_secret',
+  authCodeUrl: (oauthConfig) => {
+    return `https://www.facebook.com/v3.3/dialog/oauth?response_type=code&client_id=${oauthConfig.id}&state=${oauthConfig.state}&display=popup`;
+  },
+  tokenUrl: (oauthConfig) => {
+    return {
+      url: `https://graph.facebook.com/v3.3/oauth/access_token?client_id=${oauthConfig.id}&client_secret=${oauthConfig.secret}`,
+      options: {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+      },
+    };
+  },
+};
+
+/**
+ * Returns a object with all oauthConfiguration keys needed to get the
+ * authentication running
+ * @param {Object} providerConfig - Holds basic configuration for provider
+ * @return {Object}
+ */
+async function getOAuthConfig(providerConfig = {}) {
+  if (!Object.getOwnPropertyNames(providerConfig).length) {
+    return;
+  };
+
+  if (oauthConfig[providerConfig.provider]) {
```

Comment:
Is it possible that you forgot to add the config to oauthConfig? I couldn't find the code for it.

Reply:
I'm just checking, if the cached `oauthConfig` already holds the key of the provider I am looking for.
For example if `oauthConfig['google']` is already there, it will just return that part of the `oauthConfig`.